### PR TITLE
Add missing dec_empty_buf tests

### PR DESCRIPTION
**Pull request by k-stachowiak**
The `dec_empty_buf()` test was not called for some cipher configurations for which it is meaningful. This PR is supposed to fill the gaps.

### CODE REVIEW

File: tests/suites/test_suite_cipher.aes.data [hunk 1]
```diff
@@ -1,7 +1,11 @@
-Decrypt empty buffer
+AES-128 CBC Decrypt empty buffer
 depends_on:MBEDTLS_AES_C:MBEDTLS_CIPHER_MODE_CBC:MBEDTLS_CIPHER_PADDING_PKCS7
 dec_empty_buf:MBEDTLS_CIPHER_AES_128_CBC
 
+AES-128 GCM Decrypt empty buffer
+depends_on:MBEDTLS_AES_C:MBEDTLS_CIPHER_PADDING_PKCS7
```
Comment (gilles-peskine-arm) [inline tests/suites/test_suite_cipher.aes.data]: GCM doesn't depend on any padding, but does depend on `MBEDTLS_GCM_C`.

File: tests/suites/test_suite_cipher.aes.data [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
+dec_empty_buf:MBEDTLS_CIPHER_AES_128_GCM
```
Comment (gilles-peskine-arm) [inline tests/suites/test_suite_cipher.aes.data]: GCM belongs in `test_suite_cipher.gcm.data`, not here.

File: tests/suites/test_suite_cipher.aria.data [hunk 3]
```diff
@@ -0,0 +1,11 @@
+Aria CBC Decrypt empty buffer
+depends_on:MBEDTLS_ARIA_C:MBEDTLS_CIPHER_MODE_CBC
+dec_empty_buf:MBEDTLS_CIPHER_ARIA_128_CBC
+
+Aria CCM Decrypt empty buffer
```
Comment (gilles-peskine-arm) [inline tests/suites/test_suite_cipher.aria.data]: CCM and GCM belong in `test_suite_cipher.{ccm,gcm}.data`.

File: tests/suites/test_suite_cipher.chacha20.data [hunk 4]
```diff
@@ -1,3 +1,7 @@
+ChaCha20 Decrypt empty buffer
```
Comment (gilles-peskine-arm) [inline tests/suites/test_suite_cipher.chacha20.data]: ChaCha20 is a stream cipher, so the ciphertext and the plaintext have the same length, so this is already covered by "ChaCha20 Encrypt and decrypt 0 bytes".

**Comment by k-stachowiak** [comment]
@gilles-peskine-arm Thanks for the review. I have reworked the PR according to your instructions. I hope I haven't missed anything. The tests I omitted after eliminating the one you suggested, were either depending on the multipart decoding which was not supported for some ciphers (like CCM or NIST_KW). The XTS mode doesn't require this kind of testing as it explicitly fails with an input too short, reporting a bad input error.

File: tests/suites/test_suite_cipher.arc4.data [hunk 5]
```diff
@@ -1,3 +1,7 @@
+ARC4 Decrypt empty buffer
+depends_on:MBEDTLS_ARC4_C
+dec_empty_buf:MBEDTLS_CIPHER_ARC4_128
```
Comment (gilles-peskine-arm) [inline tests/suites/test_suite_cipher.arc4.data]: This is already covered by `enc_dec_buf` with a size of 0.

File: tests/suites/test_suite_cipher.gcm.data [hunk 6]
```diff
@@ -1,3 +1,15 @@
+AES 128 GCM Decrypt empty buffer
```
Comment (gilles-peskine-arm) [inline tests/suites/test_suite_cipher.gcm.data]: Contrary to what I wrote earlier, GCM is already covered by `enc_dec_buf` with a size of 0. In our AEAD API, passes the ciphertext-sans-tag as ciphertext and passes the tag separately, so `enc_dec_buf` with a 0-size plaintext exercises 0-size ciphertext well.

File: tests/suites/test_suite_cipher.gcm.data [hunk 7]
```diff
@@ -1,3 +1,15 @@
...
+depends_on:MBEDTLS_AES_C:MBEDTLS_GCM_C
+dec_empty_buf:MBEDTLS_CIPHER_AES_128_GCM
+
+CAMELLIA GCM Decrypt empty buffer
+depends_on:MBEDTLS_CAMELLIA_C:MBEDTLS_CIPHER_PADDING_PKCS7
```
Comment (gilles-peskine-arm) [inline tests/suites/test_suite_cipher.gcm.data]:
```suggestion
depends_on:MBEDTLS_CAMELLIA_C:MBEDTLS_GCM_C
```

File: tests/suites/test_suite_cipher.gcm.data [hunk 8]
```diff
@@ -1,3 +1,15 @@
...
+dec_empty_buf:MBEDTLS_CIPHER_CAMELLIA_128_GCM
+
+Aria GCM Decrypt empty buffer
+depends_on:MBEDTLS_ARIA_C
```
Comment (gilles-peskine-arm) [inline tests/suites/test_suite_cipher.gcm.data]:
```suggestion
depends_on:MBEDTLS_ARIA_C:MBEDTLS_GCM_C
```

File: tests/suites/test_suite_cipher.gcm.data (hunk 6 above)
Comment (gilles-peskine-arm) [reply tests/suites/test_suite_cipher.gcm.data]: Why did you remove AES-GCM but keep Camellia-GCM and ARIA-GCM?

File: tests/suites/test_suite_cipher.nist_kw.data [hunk 9]
```diff
@@ -268,4 +276,3 @@ auth_crypt_tv:MBEDTLS_CIPHER_AES_192_KWP:"21fb6600c1d34a74adee67612672593a86cf23
 KWP AES-256 wrap CAVS 17.4 FAIL COUNT 4 CLEN 32
 depends_on:MBEDTLS_AES_C:MBEDTLS_NIST_KW_C
 auth_crypt_tv:MBEDTLS_CIPHER_AES_256_KWP:"c32cb3e1e41a4b9f4de79989957866f5dd48dba38c22a6ebb80e14c84bdd9534":"":"":"c29b05c2619a58ecc1d239e7a34273cd":"":"FAIL":"":0
-
```
Comment (AndrzejKurek) [inline tests/suites/test_suite_cipher.nist_kw.data]: Did you remove an endline or an empty line here? Not sure how github shows that.

Comment (gilles-peskine-arm) [reply tests/suites/test_suite_cipher.nist_kw.data]: There was an empty line at the end of the file and that commit removed it. I think the Github rendering is unambiguous; what's hard to distinguish is trailing whitespace. Github would show a file with a missing endline differently.

**Comment by Patater** [comment]
CI on merge result is all passing. Jenkins just failed to notify GitHub with the result.

**Pull request merged**